### DISCUSSION TITLE
SMTP support

### DISCUSSION THREAD

**Discussion by Coderdude112** [anonaddy/anonaddy#182] [Feature Requests]
Hello,

In hosting other applications, I have found the need to send emails that appear to be from that application for tasks such as verification emails.

Would it be possible to implement SMTP so I can send an email from an alias?

Thanks in advance.

**Comment by jbirbal-skydom** [comment]
any updates?

**Reply by willbrowningme** [reply]
I don't have any plans to add SMTP to the service I'm afraid.

**Reply by roke-julian-lockhart** [reply]
> https://github.com/anonaddy/anonaddy/discussions/182#discussioncomment-4860395

@willbrowningme, is the current "Send" syntax exposed at https://app.addy.io/aliases not an SMTP implementation?

**Reply by willbrowningme** [reply]
The mail servers will send the email on to the correct email destination using SMTP to communicate with the destination mail server, yes.

This request is for authenticated SMTP using credentials to send directly from an alias through using an email client.

**Comment by ririko5834** [comment]
I think that it would be good to add support for SMTP and POP3/imap for aliases. Since anonaddy uses port 21 you can't use the same vps to host also custom mail server with webmail.

**Comment by luckydonald** [comment]
SMTP for sending would be really nice.

You said it's not planned to be added.
Would you be willing to look into and possibly merge MRs implementing this, though?

**Reply by willbrowningme** [reply]
It would mean a fair bit of changes on both the server and to the app and would need to provide authentication credentials for each individual alias and I'm not sure that I want that maintenance overhead.

If you know of an easy or better way to implement it then I'm open to hearing about it.

**Reply by luckydonald** [reply]
Ah, thanks for the feedback.
I forgot that actual mail stuff is handled by an standard mail thing, and not the php app providing the SMTP (i.e. as an api).

mmmh, you might be right with the complexity, though it's really a bummer as I'd need to send emails, from domains who also have addy, but I can't convince normal smtp programs to understand how to craft a forwarded mail.

**Reply by luckydonald** [reply]
Coming back to this, for my use case the login of the user, or a token for the user is all I need for now in terms of auth.
This would allow a domain to be configured in programs for software defined outgoing emails at all.

The use case is that lots of systems or programs allow you to configure smtp for outgoing mails, 
but I don't know of any program supporting converting it to a `user=domain.tld+sender@anon.io` syntax.

The idea was to plug something like https://github.com/axllent/fake-smtp in, which could then trigger the relevant anonaddy functions.

For now, adding a domain to anonaddy means it's pretty much unusable for such outgoing mails.

**Reply by luckydonald** [reply]
To add a few use cases for software which have the possibility to define SMPT credentials
- Backup software, to report about failed backups.
- Uptime Kuma, for websites going down.
- Coolify automated deployment notifications.
- Vaultwarden sign-up/recovery emails.

**Comment by bitsByB** [comment]
@willbrowningme @luckydonald

I'm enjoying the conversation and was interested to see if Addy was providing this feature. I understand this looks too much to implement.

Would you guys have some recommendation for some open source smtp, api, jmap hosted app... Not sure what are the terms.

My use case: I'm using n8n for automation and would like to avoid the ubiquitous gmail to be able to manage my emails automation.
Any idea?
Thanks a lot

**Reply by MotafokaBR** [reply]
I'm using Addy.io to foward emails from my domain to a gmail account, and I'm using MailGun as a SMTP relay to send emails from my domain.

So far it seems to be working weell with the free account, but it is just personal emails, not a marketing tool.
